jcstress-core/src/main/java/org/openjdk/jcstress/infra/grading/ReportUtils.java

```diff
@@ -290,37 +290,36 @@ public static boolean statusToPassed(TestResult result) {
 
     public static String msToDate(long ms, boolean finalDate) {
         LocalDateTime ldt = LocalDateTime.now().plus(ms, ChronoUnit.MILLIS);
-
-        String result = "";
-
         boolean overtime = ms < 0;
         if (overtime) {
             ms = -ms;
         }
+        String result = getNiceMsTimeDate(ms);
+        if (overtime) {
+            result = "overtime " + result;
+        }
+        if (!overtime) {
+            result += " left";
+        }
+        if (finalDate) {
+            result += "; at " + ldt.format(DATE_TIME_FORMATTER);
+        }
+        return result;
+    }
 
+    public static String getNiceMsTimeDate(long ms) {
+        String result = "";
         long days = TimeUnit.MILLISECONDS.toDays(ms);
         if (days > 0) {
             result += days + "d+";
             ms -= TimeUnit.DAYS.toMillis(days);
         }
         long hours = TimeUnit.MILLISECONDS.toHours(ms);
         ms -= TimeUnit.HOURS.toMillis(hours);
-
```

Comment:
Unnecessary

Reply:
Gosh. Right yo are! I missed that completely! TY!